npc/re/custom/lasagna/lasagna_npcs_1.txt

```diff
@@ -0,0 +1,2375 @@
+		next;
+		mes "[Tool Merchant]";
+		mes "^4d4dffHealer Haru's house^000000 is straight ^4d4dffto the east^000000 from here.";
+		next;
+		mes "[Tool Merchant]";
+		mes "Plants and grasses are packed in front of her house because she's busy testing medicinal herbs.";
+		mes "^4d4dffIt's the house where lots of flowerpots are aligned.^000000 You can't miss it.";
+		next;
+		mes "[Tool Merchant]";
+		mes "Is there anything you need?";
+	}
+	else {
+		mes "[Tool Merchant]";
+		mes "Welcome.";
+	}
+	next;
+	if (select( "Buy an item.", "Sell your belongings." ) == 1) {
+		mes "[Tool Merchant]";
+		mes "Do you need anything? I have basic potions, traps, magnifiers...You name it!";
+		close2;
+		callshop "tool_shop_lasagna",1;
+		end;
+	}
+	mes "[Tool Merchant]";
+	mes "Good thinking! I knew you would. you'd better sell away items that you no longer need!";
+	close2;
+	callshop "tool_shop_lasagna",2;
+	end;
+}
+-	shop	tool_shop_lasagna	-1,1750:-1,611:-1,501:-1,502:-1,503:-1,504:-1,645:-1,656:-1,713:-1,601:-1,602:-1,1065:-1
+
+lasagna,224,126,5	script	Healer Haru#doram07t	4_DR_M_02,{
+	if (lasagna_quest[0] == 7) {
+		mes "[Healer Haru]";
+		mes "I see a lot of out-of-towners these days.";
+		mes "What do you need?";
+		.@menu$[2] = "Show her the Combination Kit.";
+	}
+	else {
+		mes "[Healer Haru]";
+		mes "What's up?";
+		setarray .@menu$[0], "Somebody, help me!", "About First Aid", "About Medical Botany";
+	}
+	next;
+	switch( select( .@menu$[0], .@menu$[1], .@menu$[2] ) ) {
+	case 1:
+		if (Class == Job_Summoner) {
+			mes "[Healer Haru]";
+			mes "Oh, no...";
+			mes "It's going to be fine.";
+			npcskill "AL_HEAL",10,130,115;// 1748 hp
+			percentheal 100,100;
+			close;
+		}
+		mes "[Healer Haru]";
+		mes "Oh, dear...";
+		mes "Does it hurt a lot?";
+		mes "You need 5,000 Zeny to heal it all. Do you want me to do it?";
+		next;
+		if (select( "Say yes.", "It's too expensive." ) == 2) {
+			mes "[Healer Haru]";
+			mes "Take your time and get some rest.";
+			close;
+		}
+		if (Zeny >= 5000)// intended ?
+			Zeny = Zeny - 5000;
+		mes "[Healer Haru]";
+		mes "Thank you for using our service.";
+		npcskill "AL_HEAL",10,130,115;// 1748 hp
+		percentheal 100,100;
+		close;
+	case 2:
+		if (getskilllv("NV_FIRSTAID") > 0) {
+			mes "[Healer Haru]";
+			mes "You already know what First Aid is...";
+			mes "Please. I don't have all day.";
+			close;
+		}
+		mes "[Healer Haru]";
+		mes "Oh, you'd like to learn about First Aid! How admirable of you!";
+		mes "First Aid is a skill converting your SP into HP in an emergency.";
+		next;
+		mes "[Healer Haru]";
+		mes "The healing amount can be minimal...";
+		mes "But you can at least avoid getting stunned by a critical hit.";
+		specialeffect EF_BEGINSPELL6;
+		specialeffect EF_SPELLBREAKER;
+		next;
+		mes "[Healer Haru]";
+		mes "Click the ^4d4dffSkill^000000 button to open the Skill Window.";
+		mes "FYI, the shortcut key is ^4d4dffAlt+S^000000.";
+		getexp 0,100;
+		setquest 7722;// EMT
```

Comment:
looks like the part of code provoke another error in map-server. (typical for quests)
```
[Error]: quest_add: Character 150065 already has quest 7722.
[Debug]: Source (NPC): Healer Haru#doram07t at lasagna (224,126)
[Debug]: Function: setquest (1 parameter):
[Debug]: Data: number value=7722
[Error]: quest_update_status: Character 150065 doesn't have quest 7722.
[Debug]: (atcommand_mapmove) mapindex_name2id: Map "la_fild02" not found in index list!
```
![](https://i.imgur.com/SvrLrg0.png)